### PR TITLE
broker: content cache corner case corrections

### PR DESCRIPTION
**Pull request by chu11**
Fixes for #4378 and #4379.  Decided to try and fix both in one PR b/c they are very tightly integrated with each other.  One requires the other for a regression test, the other sort of doesn't appear without the other.

### CODE REVIEW

File: src/broker/content-cache.c
Comment (grondo) [inline src/broker/content-cache.c:787]: Commit message typo: ..."when a cache flush has not yet initiated" -- "been initiated"?

File: t/Makefile.am
Comment (grondo) [inline t/Makefile.am:297]: Commit msg: consider adding a commit body and re-wording commit subject to remove reference to github issue numbers. Not a requirement, but I do find that github issue numbers make a nice supplement to the commit message, but aren't as nice when they are the sole context. (E.g. when browsing `git log` locally you will have to separately look up issues online)

E.g. something like this (I may have gotten description wrong)
```
testsuite: add regression tests for content flush issues

Problem: No tests in the testsuite ensure that content flush is forced after
`flux content flush` nor that any dirty cache entries are flushed after a content
backing module is loaded. (Issues #4378 and #4379)

Add regression tests for these specific issues.
```

As I said, just a suggestion and it may be more personal preference, so feel free to ignore.

**Review by grondo — APPROVED** [review]
This LGTM. The solutions here seem pretty straightforward, but I do wonder if we should also wait for @garlick to approve when he returns from vacation since I'm pretty unfamiliar with this code.

I did have some suggestions inline for the commit messages, but feel free to ignore those.

**Comment by grondo** [comment]
Also, please consider retitling this PR  to `content cache corner case corrections` :stuck_out_tongue_closed_eyes:

**Comment by chu11** [comment]
re-pushed, just added a little extra comments / commit message info.

**Review by garlick — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
